netconn.go

```diff
@@ -77,7 +77,7 @@ func openConnection(uri *url.URL, tlsc *tls.Config, timeout time.Duration, heade
 			return nil, err
 		}
 
-		tlsConn := tls.Client(conn, tlsc)
+		tlsConn := tls.Client(conn, tlsConfigWithSni(uri, tlsc))
```

Comment:
Fine with this - I don't think it will break existing code. However I am going to have a think about the approach because I wonder if using a callback (say `func OnConnectionAttempt(broker *url.URL, tlsCfg *tls.Config) *tls.Config`) might provide more flexibility?

Reply:
Didn't see a reference to this callback on clone of master, if this is more aligned with idiomatic use of the library then it fits better (this can then be an SNI aware connection callback impl) - as long as it's done prior to the handshake should be fine - which looking at the name of it should be the case.

Reply:
Its not something that currently exists (would be a new callback). However it would solve solve some other issues too (currently there is only one *tls.Config but you can specify multiple brokers to attempt connections to).

Reply:
I think it might be a good idea to have such a call back exposed (`func OnConnectionAttempt`) and then we can decline the PR and not need this explicit change, allowing the consumer again to set the SNI in the callback as needed, keeping the library clean.
So I think all thats needed is an example of the proxy and a callback and the PR can then be closed allowing other consumers to implement this functionality using those tools.

Reply:
I'm happy to add the callback (might be faster as this will touch a number of files). Are you happy to do a demo? (I don't use proxies so it's not something I can really add!).

Reply:
Oh I think we got our wires crossed, I've already done the callback unless you think I missed something.
I've used the example app I've included to test against mqtt.googleapis.com with a custom proxy on localhost